src/Features/Core/Portable/CodeFixes/Configuration/ConfigureSeverityLevelCodeFixProvider.ConfigurationUpdater.cs

```diff
@@ -0,0 +1,357 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Options;
+using System;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CodeFixes.Configuration
+{
+    internal sealed partial class ConfigureSeverityLevelCodeFixProvider : ISuppressionOrConfigurationFixProvider
+    {
+        private sealed partial class ConfigurationUpdater
+        {
+            private const string DiagnosticOptionPrefix = "dotnet_diagnostic.";
+            private const string DiagnosticOptionSuffix = ".severity";
+
+            // Regular expression for .editorconfig header.
+            // For example: "[*.cs]    # Optional comment"
+            private static readonly Regex s_headerPattern = new Regex(@"\[\*([^#\[]*)\]([ ]*[;#].*)?");
+
+            // Regular expression for .editorconfig code style option entry.
+            // For example: "dotnet_style_object_initializer = true:suggestion   # Optional comment"
+            private static readonly Regex s_optionBasedEntryPattern = new Regex(@"([\w ]+)=([\w ]+):[ ]*([\w]+)([ ]*[;#].*)?");
```

Comment:
should these regexes have `start/end of line` chars in them so that they don't match in the middle of some line?  @agocke for thoughts on this as well

Reply:
The parsing logic that uses this regex reads the editorconfig file contents one line at a time, so the behavior you mentioned is already ensured.